wire-ios/Wire-iOS/Sources/UserInterface/UserProfile/Views/SecurityLevelView.swift

```diff
@@ -122,9 +109,52 @@ final class SecurityLevelView: UIView {
           iconImageView.widthAnchor.constraint(equalToConstant: 11.0),
           iconImageView.heightAnchor.constraint(equalToConstant: 11.0),
           iconImageView.centerYAnchor.constraint(equalTo: centerYAnchor),
-          iconImageView.trailingAnchor.constraint(equalTo: securityLevelLabel.leadingAnchor, constant: -10.0)
+          iconImageView.trailingAnchor.constraint(equalTo: securityLevelLabel.leadingAnchor, constant: -4.0)
         ])
     }
+
+    private func configureUpdatedCallingUI(with classification: SecurityClassification) {
+        switch classification {
+
+        case .classified:
+            securityLevelLabel.textColor = SemanticColors.Label.textSecurityEnabled
+            backgroundColor = SemanticColors.View.backgroundSecurityEnabled
+            iconImageView.setIcon(.checkmark, size: .tiny, color: SemanticColors.Icon.backgroundSecurityEnabledCheckMark)
+            topBorder.backgroundColor = SemanticColors.View.borderSecurityEnabled
+
+        case .notClassified:
+            securityLevelLabel.textColor = SemanticColors.Label.textDefaultWhite
+            backgroundColor = SemanticColors.View.backgroundSecurityDisabled
+            iconImageView.image = Asset.Images.attention.image.withTintColor(SemanticColors.Icon.foregroundCheckMarkSelected)
+            topBorder.backgroundColor = .clear
+
+        default:
+            isHidden = true
+            assertionFailure("should not reach this point")
```

Comment:
What is "this point" exactly? That we have no classification but we expect there to be one?

Reply:
Perhaps we can avoid this by defining two methods `updateForClassified`, `updateForNotClassified`, then use a switch case earlier.

Reply:
Then we would need to split with legacy/updated style again in these 2/3 functions. Or have 4/6 funcs for that.
I guess checking it with single if inside each of the 2/3 functions is not that big deal, but these amounts of `?:` operators got quite messy for sure :D

Reply:
But you mean removing assertion only? Because the case of the switch would have to stay there anyway.
I would think about replacing default with `.none` though in case of modifying the enum.

Reply:
Modified to `case .none`. It should alert as about this place when change is needed.